vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/Reasoner.java

```diff
@@ -331,6 +331,66 @@ static Reasoner getInstance() {
+	 * the query.
+	 */
+	QueryAnswersSize queryAnswerSize(PositiveLiteral query);
+
+	// TODO add examples to query javadoc
+	/**
+	 * Evaluates an atomic ({@code query}), and returns the number of implicit facts
+	 * loaded into the reasoner and the number of explicit facts materialised by the
+	 * reasoner. <br>
+	 * An answer to the query is the terms a fact that matches the {@code query}:
+	 * the fact predicate is the same as the {@code query} predicate, the
+	 * {@link TermType#CONSTANT} terms of the {@code query} appear in the answer
+	 * fact at the same term position, and the {@link TermType#VARIABLE} terms of
+	 * the {@code query} are matched by terms in the fact, either named
+	 * ({@link TermType#CONSTANT}) or anonymous ({@link TermType#NAMED_NULL}). The
+	 * same variable name identifies the same term in the answer fact. <br>
+	 * 
+	 * Depending on the state of the reasoning (materialisation) and its
+	 * {@link KnowledgeBase}, the answers can have a different {@link Correctness}
+	 * ({@link QueryResultIterator#getCorrectness()}):
+	 * <ul>
+	 * <li>If {@link Correctness#SOUND_AND_COMPLETE}, materialisation over current
+	 * knowledge base has completed, and the query answers are guaranteed to be
+	 * correct.</li>
+	 * <li>If {@link Correctness#SOUND_BUT_INCOMPLETE}, the results are guaranteed
+	 * to be sound, but may be incomplete. This can happen
+	 * <ul>
+	 * <li>when materialisation has not completed ({@link Reasoner#reason()} returns
+	 * {@code false}),</li>
+	 * <li>or when the knowledge base was modified after reasoning, and the
+	 * materialisation does not reflect the current knowledge base.
+	 * Re-materialisation ({@link Reasoner#reason()}) is required in order to obtain
+	 * complete query answers with respect to the current knowledge base.</li>
+	 * </ul>
+	 * </li>
+	 * <li>If {@link Correctness#INCORRECT}, the results may be incomplete, and some
+	 * results may be unsound. This can happen when the knowledge base was modified
+	 * and the reasoner materialisation is no longer consistent with the current
+	 * knowledge base. Re-materialisation ({@link Reasoner#reason()}) is required,
+	 * in order to obtain correct query answers.
+	 * </ul>
+	 * 
+	 *
+	 * @param query        a {@link PositiveLiteral} representing the query to be
+	 *                     answered.
+	 * @param includeNulls if {@code true}, {@link QueryResult}s containing terms of
+	 *                     type {@link TermType#NAMED_NULL} (representing anonymous
+	 *                     individuals introduced to satisfy rule existentially
+	 *                     quantified variables) will be included. Otherwise, the
+	 *                     answers will only contain the {@link QueryResult}s with
+	 *                     terms of type {@link TermType#CONSTANT} (representing
+	 *                     named individuals).
+	 * @return number of facts in the extension of the query.
```

Comment:
update the return statement in the Javadoc, now the returned value contains both the number of facts and the Correctness.